src/components/navigation/navigation-marker/navigation-marker.ts

```diff
@@ -15,42 +13,35 @@ import style from './navigation-marker.scss?lit&inline';
  * @slot - Use the unnamed slot to add `sbb-navigation-action` elements into the `sbb-navigation-marker`.
  */
 @customElement('sbb-navigation-marker')
-export class SbbNavigationMarkerElement extends SlotChildObserver(LitElement) {
+export class SbbNavigationMarkerElement extends NamedSlotListElement<SbbNavigationActionElement> {
   public static override styles: CSSResultGroup = style;
+  protected override readonly listChildTagNames = ['SBB-NAVIGATION-ACTION'];
 
   /**
    * Marker size variant.
    */
   @property({ reflect: true }) public size?: 'l' | 's' = 'l';
 
-  /**
-   * Whether the list has an active action.
-   */
-  @state() private _hasActiveAction = false;
-
-  /**
-   * Navigation action elements.
-   */
-  @state() private _actions: SbbNavigationActionElement[] = [];
+  @state() private _currentActiveAction?: SbbNavigationActionElement;
 
-  private _currentActiveAction?: SbbNavigationActionElement;
   private _navigationMarkerResizeObserver = new AgnosticResizeObserver(() =>
     this._setMarkerPosition(),
   );
 
-  protected override willUpdate(changedProperties: PropertyValues<this>): void {
-    if (changedProperties.has('size')) {
+  protected override willUpdate(changedProperties: PropertyValues<WithListChildren<this>>): void {
+    super.willUpdate(changedProperties);
+    if (changedProperties.has('size') || changedProperties.has('listChildren')) {
       this._updateMarkerActions();
     }
+    this.toggleAttribute('data-has-active-action', !!this._currentActiveAction);
   }
 
   private _updateMarkerActions(): void {
-    for (const action of this._navigationActions) {
+    for (const action of this.listChildren) {
       action.size = this.size;
     }
 
-    this._hasActiveAction = !!this._activeNavigationAction;
-    this._currentActiveAction = this._activeNavigationAction;
+    this._currentActiveAction = this.listChildren.find((action) => action.active);
```

Comment:
should we include fallback to attribute here? we had a similar case if child is not yet initialized

Reply:
Should not happen here, because it is user interaction triggered.

Reply:
You are right. I did not think about that. 😅